lib/redcord/server_scripts/shared/query_helper_methods.erb.lua

```diff
@@ -106,3 +125,58 @@ local function validate_and_parse_query_conditions(hash_tag, model, index_attrs,
   end
   return {index_sets, range_index_sets}
 end
+
+-- Validates that attributes in query are in correct order and range condition is applied only on the last attribute.
+-- Returns a table {index_key, min_string, max_string} to be used for index query.
+local function validate_and_parse_query_conditions_custom(hash_tag, model, index_name, custom_index_attrs, args)
+  if #custom_index_attrs == 0 then
+    error('Index ' .. index_name .. ' does not exist')
+  end
+  local sep = ':'
+  local i = 1
+  local j = 1
+  local min, value_string_min, query_string_min = '', '', ''
+  local max, value_string_max, query_string_max = '', '', ''
+  local is_prev_attr_query_range = false
+  while i <= #args do
+    if is_prev_attr_query_range then
+      error('Range can be applied to the last attribute of query only')
```

Comment:
is this suffcient for the case:
```
custom_index: [a, b, c]
where(a: 1, b: range_interveral)
```

Reply:
```ruby
A.where(a: 1, b: Redcord::RangeInterval.new(min: 1, max: 1), c: nil, index: :first)
A.where(a: 1, b: Redcord::RangeInterval.new(min: 1, max: 1), c: 1, d: 1, index: :first)
```
I would expect this to raise an error if we don't support the behavior

Reply:
Yes it raises error, added a test case on it